MicroPather `memset` warning

With `g++-8`, we get the following warning from the MicroPather code:
```
src/MicroPather/micropather.cpp: In member function ‘void micropather::PathCache::Reset()’:
src/MicroPather/micropather.cpp:708:42: warning: ‘void* memset(void*, int, size_t)’ clearing an object of non-trivial type ‘struct micropather::PathCache::Item’; use assignment or value-initialization instead [-Wclass-memaccess]
   memset(mem, 0, sizeof(*mem) * allocated);
                                          ^
In file included from src/MicroPather/micropather.cpp:43:
src/MicroPather/micropather.h:306:10: note: ‘struct micropather::PathCache::Item’ declared here
   struct Item
          ^~~~
```

A flawed attempt to fix this was made in PR #225. Similar related fixes were done in PR #234.

--- a/src/MicroPather/micropather.h
+++ b/src/MicroPather/micropather.h
@@ -306,15 +306,15 @@ namespace micropather
 		struct Item
 		{
 			// The key:
-			void* start = nullptr;
-			void* end = nullptr;
+			void* start;
+			void* end;
 
 			bool KeyEqual(const Item& item) const { return start == item.start && end == item.end; }
 			bool Empty() const { return start == 0 && end == 0; }
 
 			// Data:
-			void* next = nullptr;
-			float	cost = 0.0f; // from 'start' to 'next'. FLT_MAX if unsolveable.
+			void* next;
+			float	cost; // from 'start' to 'next'. FLT_MAX if unsolveable.
 
 			unsigned Hash() const
 			{

